src/javascripts/ng-admin/Crud/field/maReferenceManyField.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Edition field for a selection of elements in a list - a multiple select.
+ *
+ * @example <ma-choices-field entry="entry" field="field" value="value"></ma-choices-field>
+ */
+function maReferenceManyField($compile, ReadQueries) {
+    'use strict';
+
+    return {
+        scope: {
+            'field': '&',
+            'value': '=',
+            'entry':  '=?',
+            'datastore': '&?'
+        },
+        restrict: 'E',
+        compile: function() {
+            return {
+                pre: function(scope, element) {
+                    var field = scope.field();
+                    scope.name = field.name();
+                    scope.v = field.validation();
+                    scope.choices = [];
+
+                    var valueFieldName = field.targetEntity().identifier().name();
+                    var labelFieldName = field.targetField().name();
+
+                    scope.refreshChoices = function(search) {
+                        var referenceFields = {};
+                        referenceFields[scope.name] = field;
+
+                        return ReadQueries.getAllReferencedData(referenceFields, search)
```

Comment:
Same remark as above: this is already done in the routing `resolve` phase, you're fetching twice the same data.